src/esp/sim/Simulator.cpp

```diff
@@ -654,6 +656,13 @@ void Simulator::removeObject(const int objectID,
                              bool deleteVisualNode,
                              const int sceneID) {
   if (sceneHasPhysics(sceneID)) {
+    // delete sensor at ObjectNode if there is one
+    std::map<std::string, esp::sensor::Sensor::ptr>::iterator sensorIterator =
+        getSensorSuite().getSensors().find(std::to_string(objectID));
```

Comment:
why not cache the result of `getSensorSuite().getSensors()` in a local variable?